GVFS/GVFS.Hooks/Program.cs

```diff
@@ -417,6 +417,16 @@ private static bool ShouldLock(string[] args)
                 case "version":
                 case "web--browse":
                     return false;
+
+                /*
+                 * The following are commands that are blocked by git.exe itself.
+                 * As they are no-ops, we do not need to take the GVFS lock for these.
+                 */
+                case "gc":
+                case "prune":
+                case "repack":
+                case "submodule":

```

Comment:
What about `fsck`, `worktree`, and `update-index`?

Reply:
Of those commands, some of them are rejected by git before the pre-command hook, some are rejected after the pre-command hook, and some are only partially blocked:

### Rejected by git after pre-command hook:
- submodule
- gc

### Rejected by git before pre-command hook:
- Prene
- Repack
- fsck
- Worktree

### Only some actions blocked
- Update-Index

Looking at the git code, It looks like built-ins (checked and rejected before pre-command hook is run) are handled differently than `dashed` commands (where we run the pre-command hook).

Maybe we could instead make a change in the git side of things to bail out before calling the pre-command hook for submodule commands. //cc @benpeart